lib/iris/coords.py

```diff
@@ -2114,22 +2114,38 @@ def collapsed(self, dims_to_collapse=None):
         if np.issubdtype(self.dtype, np.str_):
             # Collapse the coordinate by serializing the points and
             # bounds as strings.
-            def serialize(x):
-                return "|".join([str(i) for i in x.flatten()])
+            def serialize(x, axis):
+                if axis is None:
+                    return "|".join(str(i) for i in x.flatten())
+
+                # np.apply_along_axis does not work with str.join, so we
+                # need to loop through the array directly. First move (possibly
```

Comment:
Is this also true for functions which call `str.join`? i.e.
```python
def np_join(x):
    return "|".join(str(i) for i in x)
```

Which seems to work with `np.apply_along_axis(np_join, 0, arr)` for the arrays I've tested with.

Reply:
Sorry I should have been clearer about what I meant by "doesn't work".  The problem comes if the output strings would have different length, some of them get truncated.  E.g.

```python
a = np.arange(25).reshape(5, 5)

print(np.apply_along_axis(np_join, 1, a))
```
```
['0|1|2|3|4' '5|6|7|8|9' '10|11|12|' '15|16|17|' '20|21|22|']
```
See [my StackOverflow question](https://stackoverflow.com/questions/52677947/join-strings-along-axis) from when I first tripped over this trying to fix `aggregated_by`.  I will update the comment.

Reply:
https://github.com/SciTools/iris/pull/5955/commits/5d5d6aee8915d5fb2f51bd61bd0281e4beb6a7fe